dropshot/src/server.rs

```diff
@@ -444,26 +482,14 @@ impl HttpsAcceptor {
                             },
                         }
                     },
-                    accept_result = tcp_listener.accept() => {
+                    accept_result = http_acceptor.accept() => {
                         let (socket, addr) = match accept_result {
                             Ok(v) => v,
                             Err(e) => {
-                                match e.kind() {
-                                    std::io::ErrorKind::ConnectionAborted => {
-                                        continue;
-                                    },
-                                    // The other errors that can be returned
-                                    // under POSIX are all programming errors or
-                                    // resource exhaustion. For now, handle
-                                    // these by no longer accepting new
-                                    // connections.
-                                    // TODO-robustness: Consider handling these
-                                    // more gracefully.
-                                    _ => {
-                                        yield Err(e);
-                                        break;
-                                    }
-                                }
+                                // The HttpAcceptor already handled transient errors.
+                                // At this point, it's a bad one.
+                                yield Err(e);
```

Comment:
As far as I can tell, the `HttpAcceptor` can't return `Err`.  Can you make it not return a `Result` and then simplify this code path?

Reply:
Yea, I can remove the possibility of errors. I left it in originally here in case you decided there was an error kind that should always bubble up as permanent, but then again, this code can be easily changed back in that case.